tests/testthat/test-show_header_names.R

```diff
@@ -0,0 +1,8 @@
+context("test-show_header_names")
+
+test_that("show_header_names works", {
+  expect_error(
+    trial %>% select(age) %>% tbl_summary() %>% show_header_names(),
```

Comment:
@jalavery , thank you for the review!!  Very good points.

1. I was using a function by accident that adds fancy quotes, instead of regular ones!  Updated!
1. Fixed the issue with the comma at the end of the list.
1. Not mentioned in your comment, but i added the arg name in the example code that is printed to make it more clear.

Here's what the new output look like:

``` r
library(gtsummary)

trial %>% select(age) %>% 
  tbl_summary() %>% 
  show_header_names()


Column Name   Column Header      
------------  -------------------
label         **Characteristic** 
stat_0        **N = 200**        
i As a usage guide, the code below re-creates the current column headers.
  modify_header(x, update = list(
    label ~ "**Characteristic**",
    stat_0 ~ "**N = 200**"
  ))
```

<sup>Created on 2020-07-01 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

I think this is ready for review again 🙌